src/libcore/iter.rs

```diff
@@ -1791,6 +1797,16 @@ impl<T, I> Iterator for Peekable<T, I> where I: Iterator<Item=T> {
 }
 
 #[stable]
+impl<T, I> ExactSizeIterator for Peekable<T, I> where I: ExactSizeIterator<Item = T> {
+    #[inline]
+    fn len(&self) -> usize {
+        // This is guarenteed to not overflow because `len()` must have been able to return a valid
+        // value before we peeked.
+        self.iter.len() + if self.peeked.is_some() { 1 } else { 0 }
+    }
+}
```

Comment:
> which must meet the ExactSizeIterator spec and must be at most (uint::MAX, Some(uint::MAX-1)).

The [`size_hint`](http://doc.rust-lang.org/std/iter/trait.Iterator.html#tymethod.size_hint) documentation does not mention `MAX - 1` and I was not under the impression that's what it was used for. I'm curious, but where'd you get the `MAX - 1` from?